website/src/controllers/api-v2/groups.js

```diff
@@ -259,11 +259,14 @@ api.attachGroup = function(req, res, next) {
     if(err) return next(err);
     if(!group) return res.json(404, {err: shared.i18n.t('messageGroupNotFound')});
 
+    var unchangedChat = _.cloneDeep(group.chat);
+
     if (!user.contributor.admin) {
-      _purgeFlagInfoFromChat(group);
+      _purgeFlagInfoFromChat(group, user);
     }
 
     res.locals.group = group;
+    res.locals.unchangedGroupChat = unchangedChat;
```

Comment:
Not sure if this is the best way to do this, but it essentially keeps a record of the unchanged chats that can be applied to the flag and like calls 

@paglias @Alys